crates/uv-client/tests/user_agent_version.rs

```diff
@@ -179,52 +179,43 @@ async fn test_user_agent_has_linehaul() -> Result<()> {
     // Deserializing Linehaul
     let linehaul: LineHaul = serde_json::from_str(uv_linehaul)?;
 
-    // Assert uv version
-    assert_eq!(uv_version, format!("uv/{}", version()));
-
-    // Assert linehaul
-    let installer_info = linehaul.installer.unwrap();
-    let system_info = linehaul.system.unwrap();
-    let impl_info = linehaul.implementation.unwrap();
-
-    assert_eq!(installer_info.name.as_deref(), Some("uv"));
-    assert_eq!(installer_info.version.as_deref(), Some(version()));
-
-    assert_eq!(system_info.name, Some(markers.platform_system));
-    assert_eq!(system_info.release, Some(markers.platform_release));
-
-    assert_eq!(impl_info.name, Some(markers.platform_python_implementation));
-    assert_eq!(
-        impl_info.version,
-        Some(markers.python_full_version.version.to_string())
-    );
-
-    assert_eq!(
-        linehaul.python,
-        Some(markers.python_full_version.version.to_string())
-    );
-    assert_eq!(linehaul.cpu, Some(markers.platform_machine));
-
-    assert_eq!(linehaul.openssl_version, None);
-    assert_eq!(linehaul.setuptools_version, None);
-    assert_eq!(linehaul.rustc_version, None);
+    // Assert linehaul user agent
+    let filters = vec![(version(), "[VERSION]")];
+    with_settings!({
+        filters => filters
+    }, {
+        // Assert uv version
+        assert_snapshot!("uv_linehaul_version", uv_version);
+        // Assert linehaul json
+        assert_json_snapshot!("uv_linehaul_json", &linehaul, {
+            ".distro" => "[distro]",
+            ".ci" => "[ci]"
+        });
+    });
 
+    // Assert distro
     if cfg!(windows) {
-        assert_eq!(linehaul.distro, None);
+        assert_json_snapshot!("uv_linehaul_distro_windows", &linehaul.distro);
```

Comment:
Can we use inline snapshots here, rather than the separate JSON files? They're pretty small.

Reply:
Is this what you had in mind? c55686c787ae1c3020b29daf2683e5da13fd5f3b